File: frontend/src/types/stats.ts
export interface UserStats {
  total_zones: number;
  public_zones: number;
  private_zones: number;
  total_records: number;
}

export interface DailyBucket {
  day: string;
  records_created: number;
}

export interface Activity {
  buckets: DailyBucket[];
}

File: backend/app/schemas/stats.py
"""Dashboard stats schemas."""

from datetime import date
from typing import Literal

from pydantic import BaseModel


class UserStatsRead(BaseModel):
    total_zones: int
    public_zones: int
    private_zones: int
    total_records: int


class DailyBucketRead(BaseModel):
    day: date
    records_created: int


class ActivityRead(BaseModel):
    buckets: list[DailyBucketRead]
